controllers/managedocs_controller.go

```diff
@@ -1085,3 +1136,51 @@ func (r *ManagedOCSReconciler) reconcileRookCephOperatorConfig() error {
 
 	return nil
 }
+
+func (r *ManagedOCSReconciler) readFederatedServiceMonitorCredentials() error {
+	secret := &corev1.Secret{}
+	selector := client.ObjectKey{
+		Namespace: r.OpenshiftMonitoringNamespace,
+		Name:      grafanaDataSourceSecretName,
+	}
+
+	err := r.UnrestrictedClient.Get(r.ctx, selector, secret)
+	if err != nil {
+		return err
+	}
+
+	type GrafanaDataSource struct {
+		BasicAuthPassword string `json:"basicAuthPassword"`
+		BasicAuthUser     string `json:"basicAuthUSer"`
+	}
+
+	type GrafanaDataSourceSecret struct {
+		DataSources []GrafanaDataSource `json:"datasources"`
+	}
+
+	prometheusConfig := secret.Data[grafanaDataSourceSecretKey]
+	datasources := GrafanaDataSourceSecret{}
+
+	err = json.Unmarshal(prometheusConfig, &datasources)
+	if err != nil {
+		return err
+	}
+	_, err = ctrl.CreateOrUpdate(r.ctx, r.Client, r.federateSecret, func() error {
+		if err := r.own(r.federateSecret); err != nil {
+			return err
+		}
+		if datasources.DataSources[0].BasicAuthUser == "" {
+			return fmt.Errorf("federate secret does not contain a Username entry")
+		}
+		r.federateSecret.Data = map[string][]byte{"Username": []byte(datasources.DataSources[0].BasicAuthUser)}
+		if datasources.DataSources[0].BasicAuthPassword == "" {
+			return fmt.Errorf("federate secret does not contain a Password entry")
+		}
+		r.federateSecret.Data["Password"] = []byte(datasources.DataSources[0].BasicAuthPassword)
+		if r.federateSecret.Data == nil {
```

Comment:
Why does the null check is done after setting Username and Password

Reply:
To make sure whether federateSecret has the data or not

Reply:
This line is guaranteed to succeed because you are defining the data field just above this line